core/http-auth-aws/src/main/java/software/amazon/awssdk/http/auth/aws/internal/signer/DefaultChecksummer.java

```diff
@@ -15,39 +15,59 @@
 
 package software.amazon.awssdk.http.auth.aws.internal.signer;
 
+import static software.amazon.awssdk.checksums.DefaultChecksumAlgorithm.SHA256;
+import static software.amazon.awssdk.http.auth.aws.internal.util.ChecksumUtil.fromChecksumAlgorithm;
+import static software.amazon.awssdk.http.auth.aws.internal.util.ChecksumUtil.readAll;
+import static software.amazon.awssdk.http.auth.aws.util.SignerConstant.X_AMZ_CONTENT_SHA256;
 import static software.amazon.awssdk.http.auth.aws.util.SignerUtils.getBinaryRequestPayloadStream;
-import static software.amazon.awssdk.http.auth.aws.util.SignerUtils.hash;
 import static software.amazon.awssdk.utils.BinaryUtils.toHex;
 
 import java.io.InputStream;
 import java.nio.ByteBuffer;
+import java.util.Collections;
 import java.util.concurrent.CompletableFuture;
 import org.reactivestreams.Publisher;
-import software.amazon.awssdk.annotations.SdkProtectedApi;
+import software.amazon.awssdk.annotations.SdkInternalApi;
 import software.amazon.awssdk.http.ContentStreamProvider;
-import software.amazon.awssdk.http.auth.aws.internal.util.DigestComputingSubscriber;
+import software.amazon.awssdk.http.SdkHttpRequest;
+import software.amazon.awssdk.http.auth.aws.internal.checksums.SdkChecksum;
+import software.amazon.awssdk.http.auth.aws.internal.io.ChecksumInputStream;
+import software.amazon.awssdk.http.auth.aws.internal.io.ChecksumSubscriber;
 import software.amazon.awssdk.http.auth.aws.signer.Checksummer;
-import software.amazon.awssdk.utils.BinaryUtils;
 
-@SdkProtectedApi
+/**
+ * The default implementation of a checksummer. By default, this will calculate the SHA256 checksum of a payload and add it as the
+ * value for the 'x-amz-content-sha256' header on the request.
+ */
+@SdkInternalApi
```

Comment:
Do we need this class? Since technically, this is covered in FlexibleChecksummer by passing just SHA256

Reply:
Nope, you're right, it can be removed.